avocado/core/loader.py

```diff
@@ -44,6 +44,52 @@
 ALL = True
 
 
+def filter_test_tags(test_suite, filter_by_tags, ignore_empty=False):
+    """
+    Filter the existing (unfiltered) test suite based on tags
+
+    The filtering mechanism is limited to INSTRUMENTED tests, that is it does
+    not apply to SIMPLE tests or any other test type.  It mean non-INSTRUMENTED
+    tests are *always* included.
+
+    :param test_suite: the unfiltered test suite
+    :type test_suite: dict
+    :param filter_by_tags: the list of tag sets to use as filters
+    :type filter_by_tags: list of comma separated tags (['foo,bar', 'fast'])
+    :param ignore_empty: if instrumented tests without tags should not be
+                         filtered out
+    :param ignore_empty: bool
+    """
+    filtered = []
+    for klass, info in test_suite:
+        if not isinstance(klass, str):  # not instrumented: no filtering
+            filtered.append((klass, info))
+            continue
+
+        test_tags = info['tags']
+        if not test_tags and ignore_empty:       # not tagged: no filtering
```

Comment:
Really? This additional state adds complexity... Not that much in the code, but yet another detail that users have to take into account.  It will make documentation lenghtier, and I fail to see a use case. 

If you guys can give me a use case, I'll feel a lot better.

Anyway, since we have two votes on this direction, I'll go ahead and implement it.